### PR TITLE
Additional text and examples about casting

### PR DESCRIPTION
**Pull request by sourcefrog**

### CODE REVIEW

File: src/doc/book/casting-between-types.md [hunk 1]
```diff
@@ -14,18 +14,24 @@ Coercion occurs in `let`, `const`, and `static` statements; in
...
+References can also be coerced to raw pointers:
 
  * `&T` to `*const T`
 
  * `&mut T` to `*mut T`
- 
- * A custom coercion using [`Deref`](deref-coercions.md)
- 
+
+Custom coercion may be defined using [`Deref`](deref-coercions.md).
```
Comment (Manishearth) [inline src/doc/book/casting-between-types.md]: "A custom coercion" or "custom coercions"

**Comment by steveklabnik** [comment]
r=me after @Manishearth 's nit

File: src/doc/book/casting-between-types.md (hunk 1 above)
Comment (sourcefrog) [inline src/doc/book/casting-between-types.md]: Thanks, fixed

**Pull request merged**